### ISSUE TITLE
Error: Unable to find latest version on GitHub

### ISSUE THREAD

**Issue by bornova** [electron-userland/electron-builder#6784]
* **Electron-Builder Version**: 23.0.3
* **Electron-Updater Version**: 5.0.1

* **Node Version**: 17.9.0

* **Electron Version**: 18.0.3
* **Electron Type (current, beta, nightly)**: Current

* **Target**: nsis

`electron-updater` 5.0.1 fails to auto update from private Github repo.  If I build my app using v5.0.1 and a newer release is available on the private repo, I now get an error after  `autoUpdater.checkForUpdatesAndNotify()`

```javascript
autoUpdater.on('error', (error) => {
  console.error(error)
})
```
logs:

> Error: Unable to find latest version on GitHub (https://api.github.com/api/v3/repos/<private-repo>), please ensure a production release exists: HttpError: 404 "method: GET url: https://api.github.com/api/v3/repos/<private-repo>/releases\n\nPlease double check that your authentication token is correct. Due to security reasons, actual status maybe not reported, but 404.\n" Headers: { "server": "GitHub.com", "date": "Mon, 11 Apr 2022 16:02:57 GMT", "content-type": "application/json; charset=utf-8", "x-oauth-scopes": "repo, workflow, write:packages", "x-accepted-oauth-scopes": "repo", "x-github-media-type": "github.v3; format=json", "x-ratelimit-limit": "5000", "x-ratelimit-remaining": "4983", "x-ratelimit-reset": "1649694314", "x-ratelimit-used": "17", "x-ratelimit-resource": "core", "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset", "access-control-allow-origin": "*", "strict-transport-security": "max-age=31536000; includeSubdomains; preload", "x-frame-options": "deny", "x-content-type-options": "nosniff", "x-xss-protection": "0", "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin", "content-security-policy": "default-src 'none'", "vary": "Accept-Encoding, Accept, X-Requested-With", "content-encoding": "gzip", "x-github-request-id": "E97B:0C3D:222588:2C9CBB:62545131" } at createHttpError (<local-install-path>\resources\app.asar\node_modules\builder-util-runtime\out\httpExecutor.js:14:12) at ElectronHttpExecutor.handleResponse (<local-install-path>\resources\app.asar\node_modules\builder-util-runtime\out\httpExecutor.js:116:20) at ClientRequest. (<local-install-path>\resources\app.asar\node_modules\builder-util-runtime\out\httpExecutor.js:82:26) at ClientRequest.emit (node:events:390:28) at SimpleURLLoaderWrapper. (node:electron/js2c/browser_init:101:6917) at SimpleURLLoaderWrapper.emit (node:events:390:28)

Looks like it may be failing to authenticate with the private token which I include in my `package.json`.  Here is the `build` part of my `package.json`:

```json
"build": {
    "publish": [
      {
        "provider": "github",
        "releaseType": "release",
        "private": true,
        "token": "<private-token>"
      }
    ],
    "win": {
      "target": "nsis"
    },
    "nsis": {
      "deleteAppDataOnUninstall": true,
      "oneClick": false
    },
    "directories": {
      "buildResources": "assets"
    }
  }
```

Reverting back to v4.6.5 of `electron-updater` fixes the issue.

**Comment by jesusrogliero** [comment]
Saludos. Casualmente me encuentro en el mismo problema

**Comment by rubenLindstrom** [comment]
Having this same issue

**Comment by kyoto44** [comment]
Same

**Comment by mozainuddin** [comment]
+, Should I downgrade?

**Comment by andrefmandrade** [comment]
> 4.6.5

This didn't work for me :(

**Comment by RoikkuTo** [comment]
I had the same problem, it looks like it had to do with the pathname of the URL :  https://api.github.com/api/v3/repos where `/api/v3` leads to nothing, which is apparently added to be compatible with the Github Enterprise API.

After looking at the code, it seems that it was missing the `host` parameter from the publish configuration. And indeed I omitted it from my `electron-builder.yml`.
So I tried to set it to `host: github.com` but I don't know why, it was causing issues and I wasn't able to publish anymore

So as a last try, I change the condition that was responsible for adding the `/api/v3` and did this : https://github.com/RoikkuTo/electron-builder/commit/72e2e71625e3e727e8f3d995a7d217252036bea7

And after that, it worked

**Comment by mmaietta** [comment]
@RoikkuTo can you please open a PR for that? I'd also recommend changing it to be clearer:
```
return host && !["github.com", "api.github.com"].includes(host) ? `/api/v3${result}` : result
```

**Comment by v360tech-sadhana** [comment]
I am facing the same issue. Please fix this issue.

**Comment by v360tech-sadhana** [comment]
> @v360tech-sadhana can you try electron-builder and electron-updater `@next` instead of `@latest`?

I moved back to electron-builder v4.6.1 and it's working fine. You can give it a try!

**Comment by italoportinho** [comment]
> @mozainuddin +, I downgrade Electron-Updater to version 4.6.5 and it works fine now.

Thanks, it worked for me.

**Comment by bornova** [comment]
> @v360tech-sadhana can you try electron-builder and electron-updater `@next` instead of `@latest`?

`@next` version (5.0.5 as of this writing) fixes the issue on my end.  When can we expect this version to hit the `@latest` channel?

**Comment by odusseys** [comment]
I am still getting this using 5.2.1 - public repo, releases are there.

**Comment by mmaietta** [comment]
Please create a sample repo of this issue. We have a few github integration tests already that always pass, so I'm not sure what config setup needs to occur for a new test repo to be used.

**Comment by tinyCoder32** [comment]
We have the same issue here, is there any solution yet?

**Comment by AdamSchinzel** [comment]
Still happening, can someone reopen the issue since it's not solved

**Comment by mmaietta** [comment]
Will reopen if: https://github.com/electron-userland/electron-builder/issues/6784#issuecomment-1277830221

**Comment by AndrewEQ** [comment]
Hi everyone, this is what I believe is happening:

1. The GitHub token generated has expired.
2. The current version of the app is making a request with the old token, you can test the request with this curl command:
```
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <YOUR-TOKEN>" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/OWNER/REPO/releases/latest
```
3. So even if you make the repo public, the update request won't work because its making the request with the old token, the way to access a public repo would be for it to make the request without the token:
```
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/OWNER/REPO/releases/latest
```

So in summary,  I don't think there's a solution for the version(s) with the expired tokens as they will be making requests to GitHub with the expired token. 

Possible solutions for future releases: set a long expiry on the GitHub token (bad if token is compromised) or make sure to make future releases with "fresher" tokens (and release often).
